File: app/src/main/java/me/miran/mhdstuff/activity/devtest/LineListActivity.java
package me.miran.mhdstuff.activity.devtest;

import android.os.Bundle;

import me.miran.mhdstuff.R;
import me.miran.mhdstuff.activity.base.BaseActivity;
import me.miran.mhdstuff.parsing.storage.IdStorage;
import me.miran.mhdstuff.parsing.storage.LineStorage;
import me.miran.mhdstuff.parsing.types.LineAlias;
import com.google.android.flexbox.FlexboxLayout;

import java.util.Comparator;
import java.util.List;

public class LineListActivity extends BaseActivity {

    private List<LineAlias> items;

    public LineListActivity() {
        super("Line test");
    }

    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_line_test);

        // TODO get only line storage?
        IdStorage.getInstanceOnUIThread((storage) -> {
            LineStorage lineStorage = storage.lineStorage();

            items = lineStorage.getAllAliases();
            items.sort(Comparator.comparing(l -> l.getSortKey(lineStorage)));

            runOnUiThread(() -> {
                FlexboxLayout lines = findViewById(R.id.line_items);

                System.out.println("ADDING "+items.size());
                for (LineAlias line : items) {
                    lines.addView(line.createLineIconView(lines, this));
                }
            });
        }, this);

    }



}

File: app/src/main/java/me/miran/mhdstuff/activity/devtest/LineListActivity.kt
package me.miran.mhdstuff.activity.devtest

import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.verticalScroll
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import me.miran.mhdstuff.R
import me.miran.mhdstuff.activity.base.KBaseActivity
import me.miran.mhdstuff.parsing.storage.IdStorage

class LineListActivity : KBaseActivity(R.string.dev_settings) {
    @Composable
    override fun CreateElements() {
        // TODO get only line storage?
        IdStorage.getInstanceOnUIThread({ storage: IdStorage ->
            val items = storage.lineStorage.allAliases

            setBaseContent {
                Container(Modifier.padding(16.dp).verticalScroll(rememberScrollState()), innerPadding = 4.dp) {
                    LineList(items)
                }
            }
        }, this)
    }


}